keyword-analyzer/src/main/java/com/example/keywordanalyzer/service/TermCountService.java

```diff
@@ -0,0 +1,52 @@
+package com.example.keywordanalyzer.service;
+
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.example.keywordanalyzer.model.entity.DocumentCount;
+import com.example.keywordanalyzer.model.entity.Post;
+import com.example.keywordanalyzer.model.entity.TermCount;
+import com.example.keywordanalyzer.repository.DocumentCountRepository;
+import com.example.keywordanalyzer.repository.TermCountRepository;
+import com.example.keywordanalyzer.repository.WordCollectionRepository;
+import com.example.keywordanalyzer.util.NlpUtil;
+
+@Service
+public class TermCountService {
+	private final TermCountRepository termCountRepository;
+	private final WordCollectionRepository wordCollectionRepository;
+	private final DocumentCountRepository documentCountRepository;
+
+	public TermCountService(TermCountRepository repository, WordCollectionRepository wordCollectionRepository,
+		DocumentCountRepository documentCountRepository) {
+		this.termCountRepository = repository;
+		this.documentCountRepository = documentCountRepository;
+		this.wordCollectionRepository = wordCollectionRepository;
+	}
+
+	@Transactional
+	public void saveTermCount(Post post) {
+		HashMap<String, Integer> termCountMap = NlpUtil.extractNoun(post.getContent());
+		Long wordCollectionId = post.getCollectionId();
+		Long postId = post.getId();
+
+		if (!wordCollectionRepository.existsById(wordCollectionId)) {
+			throw new NoSuchElementException("WordCollection not found. WordCollection Id : " + wordCollectionId);
+		}
```

Comment:
예외 처리:  
saveTermCount 메서드에서 WordCollection을 찾지 못하면 NoSuchElementException을 던지고 있다. 
더 명확한 처리를 위해 커스텀 예외를 만드는 것 권장
```suggestion
public class WordCollectionNotFoundException extends RuntimeException {
    public WordCollectionNotFoundException(Long wordCollectionId) {
        super("WordCollection not found. WordCollection Id : " + wordCollectionId);
    }
}
```

Reply:
추후 엔티티를 찾지 못한 예외를 새로 추가할 것을 고려하여 EntityNotFoundException 추가
EntityNotFoundException 상속하여 WordCollectionNotFoundException 구현